disperser/apiserver/disperse_blob_v2.go

```diff
@@ -0,0 +1,126 @@
+package apiserver
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/Layr-Labs/eigenda/api"
+	pb "github.com/Layr-Labs/eigenda/api/grpc/disperser/v2"
+	"github.com/Layr-Labs/eigenda/common"
+	v2 "github.com/Layr-Labs/eigenda/core/v2"
+	dispv2 "github.com/Layr-Labs/eigenda/disperser/common/v2"
+	"github.com/Layr-Labs/eigenda/encoding/rs"
+)
+
+func (s *DispersalServerV2) DisperseBlob(ctx context.Context, req *pb.DisperseBlobRequest) (*pb.DisperseBlobReply, error) {
+	if err := s.validateDispersalRequest(req); err != nil {
+		return nil, err
+	}
+
+	origin, err := common.GetClientAddress(ctx, s.rateConfig.ClientIPHeader, 2, true)
+	if err != nil {
+		return nil, api.NewErrorInvalidArg(err.Error())
+	}
+
+	data := req.GetData()
+	blobHeader, err := v2.NewBlobHeader(req.GetBlobHeader())
+	if err != nil {
+		return nil, api.NewErrorInternal(err.Error())
+	}
+	s.logger.Debug("received a new blob dispersal request", "origin", origin, "blobSizeBytes", len(data), "quorums", req.GetBlobHeader().GetQuorumNumbers())
+
+	// TODO(ian-shim): handle payments and check rate limits
+
+	blobKey, err := s.StoreBlob(ctx, data, blobHeader, time.Now())
+	if err != nil {
+		return nil, api.NewErrorInternal(err.Error())
+	}
+
+	return &pb.DisperseBlobReply{
+		Result:  dispv2.Queued.ToProfobuf(),
+		BlobKey: blobKey[:],
+	}, nil
+}
+
+func (s *DispersalServerV2) StoreBlob(ctx context.Context, data []byte, blobHeader *v2.BlobHeader, requestedAt time.Time) (v2.BlobKey, error) {
+	blobKey, err := blobHeader.BlobKey()
+	if err != nil {
+		return v2.BlobKey{}, err
+	}
+
+	if err := s.blobStore.StoreBlob(ctx, blobKey.Hex(), data); err != nil {
+		return v2.BlobKey{}, err
+	}
+
+	blobMetadata := &dispv2.BlobMetadata{
+		BlobHeader:  blobHeader,
+		BlobStatus:  dispv2.Queued,
+		Expiry:      uint64(requestedAt.Add(s.onchainState.TTL).Unix()),
+		NumRetries:  0,
+		BlobSize:    uint64(len(data)),
+		RequestedAt: uint64(requestedAt.UnixNano()),
+		UpdatedAt:   uint64(requestedAt.UnixNano()),
+	}
+	err = s.blobMetadataStore.PutBlobMetadata(ctx, blobMetadata)
+	return blobKey, err
+}
+
+func (s *DispersalServerV2) validateDispersalRequest(req *pb.DisperseBlobRequest) error {
+	data := req.GetData()
+	blobSize := len(data)
+	if blobSize > int(s.maxBlobSizeBytes) {
```

Comment:
one of the things auditors pointed out during the first audit was the conversions to u64 and int's. Should we conform to one, like u64, to be used?